lib/mix/test/mix/project_test.exs

```diff
@@ -138,7 +138,12 @@ defmodule Mix.ProjectTest do
 
   defp assert_proj_dir_linked_or_copied(source, target, symlink_path) do
     case :file.read_link(source) do
-      {:ok, path} -> assert path == symlink_path
+      {:ok, path} -> 
+        case :os.type do
+          # relative symlink on Windows are broken, see symlink_or_copy/2
+          {:win32, _} -> assert path == [source, '../', symlink_path] |> Path.join |> Path.expand |> String.to_charlist
```

Comment:
Also please move the `assert ...` to its own line because this one is getting quite long. :)